src/main/java/com/google/firebase/database/util/EmulatorHelper.java

```diff
@@ -0,0 +1,71 @@
+package com.google.firebase.database.util;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import com.google.firebase.database.core.Path;
+import com.google.firebase.database.core.RepoInfo;
+import com.google.firebase.database.utilities.ParsedUrl;
+import com.google.firebase.database.utilities.Utilities;
+
+public final class EmulatorHelper {
+
+  private EmulatorHelper() {
+  }
+
+  @VisibleForTesting
+  public static final String FIREBASE_RTDB_EMULATOR_HOST_ENV_VAR = "FIREBASE_RTDB_EMULATOR_HOST";
+
+  private static String getEmulatorHostFromEnv() {
+    return System.getenv(FIREBASE_RTDB_EMULATOR_HOST_ENV_VAR);
+  }
+
+  public static ParsedUrl parsedUrlForEmulator(String dbName, String emulatorUrl) {
+    RepoInfo repoInfo = new RepoInfo();
+    repoInfo.host = emulatorUrl;
+    repoInfo.namespace = dbName;
+    repoInfo.secure = false;
+    ParsedUrl parsedUrl = new ParsedUrl();
+    parsedUrl.repoInfo = repoInfo;
+    parsedUrl.path = new Path("");
+    return parsedUrl;
+  }
+
+  @VisibleForTesting
+  static String extractEmulatorUrlFromDbUrl(String suppliedDatabaseUrl) {
+    if (Strings.isNullOrEmpty(suppliedDatabaseUrl)) {
+      return null;
+    }
+    ParsedUrl parsedUrl = Utilities.parseUrl(suppliedDatabaseUrl);
+    RepoInfo repoInfo = parsedUrl.repoInfo;
+    if (repoInfo.isSecure() || repoInfo.host.endsWith(".firebaseio.com") || !suppliedDatabaseUrl
+        .contains("ns=")) {
+      return null;
+    }
+    String pathString = parsedUrl.path.isEmpty() ? "/" : parsedUrl.path.toString() + "/";
+    return String.format("http://%s%s?ns=%s", repoInfo.host, pathString, repoInfo.namespace);
+  }
+
+  public static String getEmulatorUrl(String suppliedDatabaseUrl) {
```

Comment:
A better name might be `getEmulatorHost`.

Reply:
But it returns a URL as seen below. `http://`...

Reply:
marking as resolved